tests/powermon_tests/unit/test_abstract_protocol.py

```diff
@@ -0,0 +1,108 @@
+import unittest
+import datetime
+from powermon.protocols.abstractprotocol import AbstractProtocol
+from powermon.commands.result import ResultType
+from powermon.commands.reading_definition import ResponseType
+from powermon.commands.reading_definition import ReadingType
+from powermon.commands.parameter import ParameterType
+from powermon.commands.parameter import ParameterFormat
+
+class test_protocol_pi30(unittest.TestCase):
+    def test_add_command_definitions(self):
+        protocol = AbstractProtocol()
+        #Just here for comparison
+        pbt_old_command_definitions = {
+            "PBT": {
+                "name": "PBT",
+                "description": "Set Battery Type",
+                "help": " -- examples: PBT00 (set battery as AGM), PBT01 (set battery as FLOODED), PBT02 (set battery as USER)",
+                "response_type": ResultType.ACK,
+                "response": [[0, "Command execution", ResponseType.ACK, {"NAK": "Failed", "ACK": "Successful"}]],
+                "test_responses": [b"(NAK\x73\x73\r", b"(ACK\x39\x20\r",],
+                "regex": "PBT(0[012])$",
+            },
+        }
+        pbt_command_definition_new = {
+            "PBT": {
+                "name": "PBT",
+                "description": "Set Battery Type",
+                "help": " -- examples: PBT00 (set battery as AGM), PBT01 (set battery as FLOODED), PBT02 (set battery as USER)",
+                "result_type": ResultType.ACK,
+                "reading_definitions": [{"index":0, "decription":"Command execution", "reading_type":ReadingType.ACK, "response_type":ResponseType.ACK}],
```

Comment:
I think this and the line below are redundant and can be determined by ResultType.ACK